driver-core/src/main/com/mongodb/internal/connection/InternalConnectionFactory.java

```diff
@@ -17,7 +17,22 @@
 package com.mongodb.internal.connection;
 
 import com.mongodb.connection.ServerId;
+import org.bson.types.ObjectId;
 
 interface InternalConnectionFactory {
-    InternalConnection create(ServerId serverId);
+    default InternalConnection create(ServerId serverId) {
```

Comment:
This method is still used by the server monitor, which doesn't care about generations.  So we supply a default implementation.